### PR TITLE
chore: Track opentelemetry-proto as a submodule, currently tracking 0.19.0

### PR DESCRIPTION
**Pull request by spencergilbert**

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
🎉

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "lib/opentelemetry-proto/src/proto/opentelemetry-proto"]
```
Comment (neuronull) [inline .gitmodules:1]: should we be doing this for `lib/loki-logproto` as well?

File: lib/opentelemetry-proto/Cargo.toml [hunk 2]
```diff
@@ -4,6 +4,7 @@ version = "0.1.0"
 authors = ["Vector Contributors <vector@datadoghq.com>"]
 edition = "2021"
 publish = false
+license = "MPL-2.0"
```
Comment (spencergilbert) [inline lib/opentelemetry-proto/Cargo.toml:7]: Adding such that we can remove a license tool override.

File: lib/opentelemetry-proto/build.rs
Comment (spencergilbert) [inline lib/opentelemetry-proto/build.rs:1]: Just alphabetized

File: lib/opentelemetry-proto/Cargo.toml (hunk 2 above)
Comment (spencergilbert) [inline lib/opentelemetry-proto/Cargo.toml:7]: Adding such that we can remove a license tool override.

**Pull request closed without merging**